src/Adapter/MSTest.TestAdapter/Execution/TestMethodRunner.cs

```diff
@@ -306,6 +311,20 @@ private bool ExecuteDataSourceBasedTests(List<TestResult> results)
                 foreach (UTF.ITestDataSource testDataSource in testDataSources)
                 {
                     isDataDriven = true;
+
+                    if (testDataSource is ITestDataSourceIgnoreCapability { Ignore: { } ignoreReason })
+                    {
+                        results.Add(new()
+                        {
+                            // This is closest to ignore. This enum doesn't have a value specific to Ignore.
+                            // It may be a better idea to add a value there, but the enum is public and we need to think more carefully before adding the API.
+                            // For now, TestResultExtensions.ToUnitTestResults method will convert this to Ignored value of ObjectModel enum when IgnoreReason is non-null.
+                            Outcome = UTF.UnitTestOutcome.NotRunnable,
```

Comment:
It's probably good to have more states but I will have a closer look later.

Reply:
Sure.